src/cryptonote_core/blockchain.cpp

```diff
@@ -1766,7 +1766,8 @@ bool Blockchain::handle_alternative_block(const block& b, const crypto::hash& id
   bool service_node_checkpoint = false;
   if (!m_checkpoints.is_alternative_block_allowed(get_current_blockchain_height(), block_height, &service_node_checkpoint))
   {
-    if (!service_node_checkpoint || b.major_version >= cryptonote::network_version_13_enforce_checkpoints)
+    // NOTE: The only case where we want to override this is_alternative_block_allowed failing is when you are in the soft forking period
+    if (!(service_node_checkpoint && b.major_version <= cryptonote::network_version_12_checkpointing))
```

Comment:
Is this condition supposed to be different than before?  It looks logically identical, but harder to read.

Reply:
Hmmmm. It looks logically identical, and it seems to be after thinking about it. But I changed it because I found the initial way I wrote it confusing. Perhaps its clearer like this

```
bool soft_fork_checkpoint = (b.major_version <= cryptonote::network_version_12_checkpointing && service_node_checkpoint);
if (!soft_fork_checkpoint)
```

Reply:
Sure, that helps.